src/Stratis.SmartContracts.IntegrationTests/ContractPoAStressTests.cs

```diff
@@ -53,5 +61,75 @@ public void MaximumCreateTransactionsInABlock()
+
+            // Nodes are syncing together...
+            this.mockChain.MineBlocks(1);
+
+            // Node 1 loses connection to the others
+            foreach (var peer in node1.CoreNode.FullNode.ConnectionManager.ConnectedPeers.ToList())
+            {
+                peer.Disconnect("For Testing");
+            }
+            TestBase.WaitLoop(() => !node1.CoreNode.FullNode.ConnectionManager.ConnectedPeers.Any());
+
+            // Node 2 sends contract tx and creates its own chain with node 3
+            ContractCompilationResult compilationResult = ContractCompiler.CompileFile("SmartContracts/Auction.cs");
+            Assert.True(compilationResult.Success);
+            BuildCreateContractTransactionResponse response = node2.SendCreateContractTransaction(compilationResult.Compilation, 0, gasLimit:SmartContractFormatLogic.GasLimitMaximum);
+            node3.WaitMempoolCount(1);
+            node3.CoreNode.MineBlocksAsync(2).GetAwaiter().GetResult();
+            TestBase.WaitLoop(() =>node2.CoreNode.FullNode.ChainIndexer.Height == startingHeight + 3);
+
+            // Node gets a refund utxo in the coinbase
+            var unspents = node2.SpendableTransactions.ToList();
+            Assert.True(unspents[1].Transaction.IsCoinBase);
+
+            // Refund utxo is used to build a new transaction
+            BuildCreateContractTransactionResponse response2 = node2.SendCreateContractTransaction(compilationResult.Compilation, 0, gasLimit: 15000uL, outpoints: new List<OutpointRequest>
+            {
+                new OutpointRequest
+                {
+                    Index = unspents[1].Transaction.Index,
+                    TransactionId = unspents[1].Transaction.Id.ToString()
+                }
+            });
+            Transaction tx = node1.CoreNode.FullNode.Network.CreateTransaction(response2.Hex);
+            Assert.Equal(unspents[1].Transaction.Id, tx.Inputs[0].PrevOut.Hash);
+            node2.WaitMempoolCount(1);
+            node3.WaitMempoolCount(1);
+
+            // Other node mines far ahead
+            node1.CoreNode.MineBlocksAsync(5).GetAwaiter().GetResult();
+            Assert.True(node2.CoreNode.FullNode.ChainIndexer.Height == startingHeight + 3);
+            Assert.True(node3.CoreNode.FullNode.ChainIndexer.Height == startingHeight + 3);
+            Assert.True(node1.CoreNode.FullNode.ChainIndexer.Height == startingHeight + 6);
+
+            // Reconnect nodes.
+            TestHelper.Connect(node1.CoreNode, node2.CoreNode);
+            TestHelper.Connect(node1.CoreNode, node3.CoreNode);
+
+            // 2 and 3 will reorg to 1's chain.
+            TestBase.WaitLoop(() => node2.CoreNode.FullNode.ChainIndexer.Height == startingHeight + 6);
+            TestBase.WaitLoop(() => node3.CoreNode.FullNode.ChainIndexer.Height == startingHeight + 6);
+
+            // Lets give some time to the nodes to try and sort themselves out.
+            Thread.Sleep(5_000);
```

Comment:
Generally the TestHelper's `AreNodesSynced` method should wait until all the nodes are on the same height.